tests/test_generate_manifest.py

```diff
@@ -0,0 +1,101 @@
+from os.path import join, dirname, abspath
+from unittest.mock import Mock, patch
+from tests.test_common import TestEPP, NamedMock,PropertyMock
+from scripts.generate_manifest import GenerateManifest
+
+
+class TestGenerateManifest(TestEPP):
+    udf = {'[Plates]Sample Name': 'A',
+           '[Plates]Container Name': 'B',
+           '[Plates]Well': 'C',
+           '[Plates]Project ID': 'D',
+           '[Plates][Sample UDF]Column': 'E',
+           '[Plates]Starting Row': 1,
+           '[Tubes]Sample Name': 'A',
+           '[Tubes]Container Name': 'B',
+           '[Tubes]Well': 'C',
+           '[Tubes]Project ID Well': 'D1',
+           '[Tubes][Sample UDF]Column': 'E',
+           '[Tubes]Starting Row': 1,
+           '[SGP]Sample Name': 'A',
+           '[SGP]Container Name': 'B',
+           '[SGP]Well': 'C',
+           '[SGP]Project ID Well': 'D1',
+           '[SGP][Sample UDF]Column': 'E',
+           '[SGP]Starting Row': 1}
+
+
+    patch_process_plate = patch.object(
+        GenerateManifest,
+        'process',
+        new_callable=PropertyMock(return_value=
```

Comment:
These mocked process would be easier to read if generated with `FakeEntityMaker`

Reply:
Updated (also update test_common to add new container types)